prov/rxm/src/rxm_init.c

```diff
@@ -66,33 +63,17 @@ int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 		if (hints->caps & (FI_MSG | FI_TAGGED))
 			core_info->caps |= FI_RMA;
 
-		/* No fi_info modes apart from those handled earlier in
-		 * this function can be passed along to the core provider */
-
-		if (hints->domain_attr) {
-			if (FI_VERSION_GE(version, FI_VERSION(1, 5))) {
-				/* Allow only those mr modes that can be
-				 * passed along to the core provider */
-				core_info->domain_attr->mr_mode |=
-					hints->domain_attr->mr_mode &
-					OFI_MR_BASIC_MAP;
-			} else if (ofi_rma_target_allowed(hints->caps)) {
-				core_info->domain_attr->mr_mode =
-					hints->domain_attr->mr_mode;
-			}
+		if (hints->domain_attr)
 			core_info->domain_attr->caps |= hints->domain_attr->caps;
-		}
+
 		if (hints->tx_attr) {
 			core_info->tx_attr->msg_order = hints->tx_attr->msg_order;
 			core_info->tx_attr->comp_order = hints->tx_attr->comp_order;
 		}
 	}
 
-	/* Remove caps that RxM can handle */
 	core_info->rx_attr->msg_order &= ~FI_ORDER_SAS;
 
-	core_info->ep_attr->type = FI_EP_MSG;
-
 	return 0;
 }
```

Comment:
Most of the comments in this function are unnecessary, as they are implied by the purpose of the call.